### PR TITLE
dd-opentracing-cpp: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

**Comment by RubenRBS** [comment]
The last grpc issue lost some PR jobs, I've retriggered them now

File: recipes/dd-opentracing-cpp/all/conanfile.py [hunk 1]
```diff
@@ -22,99 +29,80 @@ class DatadogOpenTracingConan(ConanFile):
...
+                "Datadog-opentracing requires C++14. Your compiler is unknown. Assuming it supports C++14."
+            )
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if self.settings.compiler == "clang" and Version(self.settings.compiler.version) < "4.0":
```
Comment (uilianries) [inline recipes/dd-opentracing-cpp/all/conanfile.py:83]: Could you please add a commend like `# INFO: ` why is it needed now?

File: recipes/dd-opentracing-cpp/all/conanfile.py [hunk 2]
```diff
@@ -22,99 +29,80 @@ class DatadogOpenTracingConan(ConanFile):
...
-            "DD_OPENTRACING_SHARED" if self.options.shared else "DD_OPENTRACING_STATIC"
-        )
+        self.cpp_info.libs = ["dd_opentracing"]
+        self.cpp_info.defines.append("DD_OPENTRACING_SHARED" if self.options.shared else "DD_OPENTRACING_STATIC")
         if self.settings.os in ("Linux", "FreeBSD"):
-            self.cpp_info.components["dd_opentracing"].system_libs.append("pthread")
-
-        # TODO: to remove in conan v2 once cmake_find_package_* generators removed.
-        #       Do not support these names in CMakeDeps, it was a mistake, upstream doesn't export targets
-        self.cpp_info.names["cmake_find_package"] = "DataDogOpenTracing"
```
Comment (uilianries) [inline recipes/dd-opentracing-cpp/all/conanfile.py:112]: I see no new properties, but removing old components. Was it wrong before?

**Review by uilianries — COMMENTED** [review]
Looking good, just few questions.

File: recipes/dd-opentracing-cpp/all/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/dd-opentracing-cpp/all/conanfile.py:83]: I don't remember the exact reason anymore. I tested the older versions in a Docker container and that was the latest supported version.

File: recipes/dd-opentracing-cpp/all/conanfile.py (hunk 2 above)
Comment (valgur) [reply recipes/dd-opentracing-cpp/all/conanfile.py:112]: I based it on the comment above:
```
# TODO: to remove in conan v2 once cmake_find_package_* generators removed.
#       Do not support these names in CMakeDeps, it was a mistake, upstream doesn't export targets
```

File: recipes/dd-opentracing-cpp/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/dd-opentracing-cpp/all/conanfile.py:83]: It still miss information, I see no relation in the upstream. Please, remove it.

File: recipes/dd-opentracing-cpp/all/patches/1.3.0/0001-Find-packages-setup.patch [hunk 3]
```diff
@@ -25,25 +21,25 @@
...
  install(DIRECTORY include/datadog DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
  file(GLOB DD_OPENTRACING_SOURCES "src/*.cpp")
-@@ -58,29 +53,24 @@ else()
+@@ -58,29 +53,28 @@ else()
  endif()
  
  # Outputs
 -set(DATADOG_LINK_LIBRARIES ${OPENTRACING_LIB} ${CURL_LIBRARIES} ${ZLIB_LIBRARIES} Threads::Threads)
-+set(DATADOG_LINK_LIBRARIES OpenTracing::OpenTracing CURL::CURL msgpack::msgpack ZLIB::ZLIB Threads::Threads)
++set(DATADOG_LINK_LIBRARIES OpenTracing::OpenTracing CURL::libcurl ${msgpack_LIBRARIES} ZLIB::ZLIB Threads::Threads)
```
Comment (uilianries) [inline recipes/dd-opentracing-cpp/all/patches/1.3.0/0001-Find-packages-setup.patch:29]:
```suggestion
+set(DATADOG_LINK_LIBRARIES OpenTracing::OpenTracing CURL::libcurl msgpack-cxx ZLIB::ZLIB Threads::Threads)
```

File: recipes/dd-opentracing-cpp/all/patches/1.3.0/0001-Find-packages-setup.patch [hunk 4]
```diff
@@ -25,25 +21,25 @@
...
++include_directories(${msgpack_INCLUDE_DIRS})
```
Comment (uilianries) [inline recipes/dd-opentracing-cpp/all/patches/1.3.0/0001-Find-packages-setup.patch:30]:
```suggestion
```

**Review by danimtb — APPROVED** [review]

**Review by DS-Alex-AG — APPROVED** [review]

**Comment by jjcasmar** [comment]
Can this be merged? Or what is missing? I was about to write my own v2 recipe

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**